### PR TITLE
Add check for google_compute_router bgp.advertise_mode.

### PR DESCRIPTION
**Pull request by xingao267**
Fixes terraform-providers/terraform-provider-google#5382

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added check on `google_compute_router` for non-empty advertised_groups or advertised_ip_ranges values when advertise_mode is DEFAULT in the bgp block.
```

### CODE REVIEW

File: templates/terraform/constants/router.go.erb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// customizeDiff func for additional checks on google_compute_router properties:
+func resourceComputeRouterCustomDiff(diff *schema.ResourceDiff, meta interface{}) error {
+
+	block := diff.Get("bgp.0").(map[string]interface{})
+	advertiseMode := block["advertise_mode"]
+	advertisedGroups := block["advertised_groups"].([]interface{})
+	advertisedIPRanges := block["advertised_ip_ranges"].([]interface{})
+
+	if advertiseMode == "DEFAULT" && len(advertisedGroups) != 0 {
+		return fmt.Errorf("Error in bgp: advertised_groups cannot be specified when using advertise_mode DEFAULT")
```
Comment (xingao267) [inline templates/terraform/constants/router.go.erb]: I wanted to use errors.New() here but I couldn't figure out how to add a new import in the final combined go file.

Comment (danawillow) [reply templates/terraform/constants/router.go.erb]: we run goimports as part of the magician, so you shouldn't have to worry about adding it explicitly. However, we use fmt.Errorf all the time so I wouldn't worry too much about it. Are there any specific advantages of errors.New that make you prefer it?

File: templates/terraform/constants/router.go.erb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2019 Google Inc.
```
Comment (danawillow) [inline templates/terraform/constants/router.go.erb]:
```suggestion
	# Copyright 2020 Google Inc.
```

File: templates/terraform/constants/router.go.erb (hunk 1 above)
Comment (xingao267) [reply templates/terraform/constants/router.go.erb]: Ok. I'll leave it as fmt.Errorf. I usually use fmt.Errorf when I need to format string in the error message and errors.New when i don't.

**Comment by xingao267** [comment]
@danawillow Thanks for the review.

Thanks for fixing the changelog message. I missed the separate repo part.

**Pull request merged**